http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/http2/MultiplexedChannelRecord.java

```diff
@@ -248,9 +249,13 @@ public Channel getConnection() {
         return connection;
     }
 
-    public boolean claimStream() {
+    private boolean claimStream() {
         lastReserveAttemptTimeMillis = System.currentTimeMillis();
         for (int attempt = 0; attempt < 5; ++attempt) {
+            if (!connection.isActive()) {
+                return false;
+            }
+
```

Comment:
Do we know that this actually helps, or are we just guessing?

Reply:
As discussed offline, the check does not cover for all cases and we need to figure out a better way to reduce the race condition. Removed it for now.